src/ui/public/visualize/__tests__/visualize.js

```diff
@@ -109,13 +108,14 @@ describe('visualize directive', function () {
 
   describe('request handler', () => {
 
-    const requestHandler = sinon.stub().returns(new Promise(_.noop));
-    // Create utility property `lastParams` to quickly access last passed params
-    Object.defineProperty(requestHandler, 'lastParams', {
-      get() {
-        return requestHandler.lastCall.args[1];
-      }
-    });
+    const requestHandler = sinon.stub().resolves();
+
+    /**
+     * Asserts that a specific parameter had a specific value in the last call to the requestHandler.
+     */
+    function assertParam(obj) {
+      sinon.assert.calledWith(requestHandler, sinon.match.any, sinon.match(obj));
```

Comment:
__optional nit:__ I usually prefer `calledWithExactly`, but it may not be important in every case, so up to you.

Reply:
I think I make it dependent on the actual check and method I want to check. Since I really only care about the second argument here, I would leave it like that. So e.g. if we extend the request handler to have a third argument in the end (which might be possible), all of those tests would fail otherwise, even though actually the actual contract that we are testing is still working as expected, that's why I would go with `calledWith` instead of `calledWithExactly` for this specific tests.